tap_s3_csv/sync.py

```diff
@@ -117,20 +120,26 @@ def handle_file(config, s3_path, table_spec, stream, extension, file_handler=Non
             # so we need to pass in fieldnames. First thread needs to handle first row if table_spec.has_header == True in order to avoid
             # having first row parsed as record when it's actually header. Set handle_first_row param for PreprocessStream to True
             # for this case so that the file/stream pointer is moved to skip first row.
+
+            # with import file copy for sftp, catalog is different from csv one, column_order is not present, so using cols_from_metadata
             file_handle = preprocess.PreprocessStream(
                 file_handle, table_spec, start_byte == 0 and table_spec.get('has_header', True))
-            fieldnames = stream['column_order']
+
+            fieldnames = cols_from_metadata
```

Comment:
it will not fail for correct config files. But I will change to throw exception instead of returning empty array.

Reply:
if statement in line 134 depends on the array length, so I guess we could check for column_order as fallback here, then throw error if column_order does not exist instead of changing the return type?

Reply:
changes done accordingly